src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -361,16 +361,17 @@ internal FunctionLoadRequest GetFunctionLoadRequest(FunctionMetadata metadata, M
         internal void LoadResponse(FunctionLoadResponse loadResponse)
         {
             _functionLoadRequestResponseEvent?.Dispose();
-            _workerChannelLogger.LogDebug("Received FunctionLoadResponse for functionId:{functionId}", loadResponse.FunctionId);
+            string functionName = _functions.Single(m => m.GetFunctionId().Equals(loadResponse.FunctionId, StringComparison.OrdinalIgnoreCase)).Name;
```

Comment:
Could any of this ever throw? I'd recommend doing SingleOrDefault() just in case -- and using something like `[null]` if it's null. Then we could debug that if it ever happens -- but at least it won't blow up.

Same with `GetFunctionId()` -- will that ever throw or be null?

I'm very cautious nowadays :-)

Reply:
Updated to SingleOrDefault(). Thanks!  Safe to use `GetFunctionId()` this is accesses multiple place before this code is executed.